src/clj/fluree/db/conn/file.cljc

```diff
@@ -31,14 +30,12 @@
                      data
                      (json-ld/normalize-data data))
           bytes    (bytes/string->UTF8 json)
-          hash     (crypto/sha2-256 bytes :hex)
           type-dir (name data-type)
-          path     (str alias
-                        (when branch (str "/" branch))
-                        (str "/" type-dir "/")
-                        hash ".json")
```

Comment:
Are we going to write a migration to handle this breaking change? I really regretted putting this suffix here but didn't remove it when I did the storage refactor in order to maintain compatibility. I think we do need to remove it in order to have a proper storage protocol, though.

Reply:
I'm going to roll it in to the sid migration if it's necessary since that migration hasn't been released yet and no one that I know of has run it.

Reply:
I think the .json was useful because we knew the encoding of the file (as a side benefit, it was also useful to double-click on commit or data files and have the editor pop open the file without extra steps). The file store is the most obvious place where this surfaces... because IPFS actually embeds the encoding into its format, and memory store doesn't really matter. I don't think anyone touched S3... so not sure if it ever did the right thing.

The nice thing about JSON-LD is the standard defines all three of (a) serialization, (b) encoding, (c) hashing algo... but if we want to support other formats we need to consider how to store potentially all three of those pieces of metadata somehow.

I don't see us favoring anything but JSON-LD for commit and data files, but I could see a strong argument for storing index files in a more compact/speedy format. We could again rely on file extension here (e.g. `.avro`) or we'll need to move to some existing or custom format where this is encoded into either the file name or some binary metadata.

Reply:
I could see us eventually storing a more compact representation even for commit and especially data files at some point because there's a lot of opportunity for compression. There are definitely advantages to using the suffix when we know the format, but I don't think we can rely on the format being the same, especially since we could use the same storage implementation to store commit files, data files, and indexes, which you could argue could each be optimally stored in a different format. I think baking in the json suffix ties our hands

I think the storage implementation is most flexible if it's only concerned with putting and pulling bytes too/from disk. We can rely on the things built on top of it to understand what format those bytes are and how to interpret them, and those higher level components can worry about performing whatever (de)serialization required before things are written and after they're read from storage.

Reply:
My comment is more along the lines that if storage is just pushing/pulling bytes, then that's probably all it should do. Right now it is it is determining hash algorithm, file name and encoding.

Unfortunately IPFS combines all of these things into one protocol... which sort of means we have no choice but to also embed these things into every storage protocol... so it can't be a simple as pushing/pulling bytes.

But now our FS storage no longer does this. It used to (using the .json suffix) but now doesn't. It seems we would need a mechanism to do this if we want to support multiple file formats.

I'm not saying `json` can't be passed as an argument to storage instead of hard coded like it was, and storage (a) might append it as suffix `.json`, while storage (b) might encode it in a binary header (like IPFS does), or storage (c) might use multi-format and encode it into the file name characters.

Reply:
Ok, I think I understand what you're getting at, and I think that makes sense. I think we can add back the suffix in the file store for now and figure out what to do if/when we start supporting multiple formats.